tests/admin_views/test_autocomplete_view.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import unicode_literals
+
+import json
+import sys
+
+from django.contrib.admin import site
+from django.contrib.admin.tests import AdminSeleniumTestCase
+from django.contrib.admin.views.autocomplete import (
+    AutocompleteJsonView, AutocompletePage, AutocompletePaginator,
+)
+from django.contrib.auth.models import Permission, User
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+from django.test import RequestFactory, TestCase, override_settings
+from django.urls import reverse, reverse_lazy
+from django.utils.six import text_type
+
+from .admin import AnswerAdmin, QuestionAdmin
+from .models import Answer, Question
+from .tests import AdminViewBasicTestCase
+
+site.register(Question, QuestionAdmin)
+site.register(Answer, AnswerAdmin)
+
+
+class AutocompleteJsonViewTest(AdminViewBasicTestCase):
+    url = reverse_lazy('admin:autocomplete')
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = User.objects.create_user(username='user', password='secret', email='user@example.com')
```

Comment:
It seems both times this user is used, it gets `is_staff=True`. Maybe move there into the setup and call it "staff_user"?